src/Utilities/hooks/usePromiseQueue.js

```diff
@@ -0,0 +1,33 @@
+import { useCallback, useState } from 'react';
+import pAll from 'p-all';
+
+const DEFAULT_CONCURRENT_PROMISES = 2;
+
+const usePromiseQueue = (limit = DEFAULT_CONCURRENT_PROMISES) => {
+    const [isResolving, setIsResolving] = useState(false);
```

Comment:
How about merging these 2 states? This would make the code cleaner and also as far as I remember state updates in a promise do not get batched (reference: https://stackoverflow.com/questions/53048495/does-react-batch-state-update-functions-when-using-hooks).

Reply:
nice suggestion, done in the last commit.